### PR TITLE
feat(google-maps): Add MapPolygon component

### PR DESCRIPTION
**Pull request by mbehrlich**
Add MapPolygon component to Google Maps, allowing polygons to be added
to a Google Map.

### CODE REVIEW

**Comment by jelbourn** [comment]
Marking this as blocked since it will need to go in after we release v9

**Comment by jelbourn** [comment]
@mbehrlich actually I tried the demo locally and there seems to be an error when enabling the polygon:
```
core.umd.js:4194 ERROR TypeError: Cannot read property 'getPaths' of undefined
```

**Comment by mbehrlich** [comment]
Does this still need to be blocked since it's being merged into google-maps-next and not master?

File: src/dev-app/google-map/google-map-demo.html [hunk 1]
```diff
@@ -28,4 +29,13 @@
     <input id="editable-polyline-checkbox" type="checkbox" (click)="toggleEditablePolyline()">
   </div>
 
+  <div>
+    <label for="polygon-checkbox">Toggle Polygon</label>
+    <input id="polygon-checkbox" type="checkbox" (click)="togglePolygonDisplay()">
```
[the remark this answers is not in the document]
Comment (mbehrlich) [reply src/dev-app/google-map/google-map-demo.html]: Done

**Comment by mbehrlich** [comment]
The type error occurs if you click "toggle editable polygon" when the polygon itself is not displayed. As far as @angular/google-maps is concerned, that's working as intended. To fix this in the demo, I've disabled the editable toggle when the polygon (or polyline) itself is not being displayed.

**Pull request merged**